datahub/dnb_api/test/test_utils.py

```diff
@@ -554,3 +558,74 @@ def test_get_company_dnb_service_request_error(
         assert str(excinfo.value) == expected_message
         assert len(caplog.records) == 1
         assert caplog.records[0].getMessage() == expected_message
+
+
+class TestRollbackDNBCompanyUpdate:
+    """
+    Test rollback_dnb_company_update utility function.
+    """
+
+    @pytest.mark.parametrize(
+        'fields, expected_fields',
+        (
+            (None, ALL_DNB_UPDATED_MODEL_FIELDS),
+            (['name'], ['name']),
+        ),
+    )
+    def test_rollback(
+        self,
+        formatted_dnb_company,
+        fields,
+        expected_fields,
+    ):
+        """
+        Test that rollback_dnb_company_update will roll back all DNB fields.
+        """
+        with reversion.create_revision():
+            company = CompanyFactory(duns_number=formatted_dnb_company['duns_number'])
+
+        original_company = Company.objects.get(id=company.id)
+
+        update_company_from_dnb(
+            company,
+            formatted_dnb_company,
+            update_descriptor='foo',
+        )
+
+        rollback_dnb_company_update(company, 'foo', fields_to_update=fields)
+
+        company.refresh_from_db()
+        for field in expected_fields:
+            assert getattr(company, field) == getattr(original_company, field)
+
+        latest_version = Version.objects.get_for_object(company)[0]
+        latest_version.revision.comment = 'Reverted D&B update from: foo'
+
+    @pytest.mark.parametrize(
+        'update_comment, error_message',
+        (
+            ('foo', 'Revision with comment: foo is the base version.'),
+            ('bar', 'Revision with comment: bar not found.'),
+        ),
+    )
+    def test_rollback_error(
+        self,
+        formatted_dnb_company,
+        update_comment,
+        error_message,
+    ):
+        """
+        Test that rollback_dnb_company_update will fail with the given error
+        message when there is an issue in finding the version to revert to.
+        """
+        company = CompanyFactory(duns_number=formatted_dnb_company['duns_number'])
+
+        update_company_from_dnb(
+            company,
+            formatted_dnb_company,
+            update_descriptor='foo',
+        )
+
+        with pytest.raises(RevisionNotFoundError) as e:
```

Comment:
[Would suggest `excinfo` instead of `e` (it's an instance of ExceptionInfo)}(https://docs.pytest.org/en/latest/assert.html?highlight=raises#assertions-about-expected-exceptions):

```suggestion
        with pytest.raises(RevisionNotFoundError) as excinfo:
```